code/modules/mob/living/living.dm

```diff
@@ -917,9 +917,15 @@ Thanks.
 	//putting out a fire
 		if(CM.on_fire && CM.canmove)
 			CM.fire_stacks -= 5
-			CM.weakened = 5
+			CM.SetWeakened(3)
+			playsound(CM.loc, 'sound/effects/bodyfall.ogg', 50, 1)
 			CM.visible_message("<span class='danger'>[CM] rolls on the floor, trying to put themselves out!</span>",
 							   "<span class='warning'>You stop, drop, and roll!</span>")
+
+			for(var/i in list(1, 4, 2, 8, 1, 4, 2, 8, 1, 4))
```

Comment:
Well I can't get it to work with that method. I'll ask at IRC when I get some free time